eclipse-theia/theia
# Add existence check before adding widget

Signed-off-by: Ken Aoki <ken3574.25@gmail.com>



#### What it does

Fixes #8511 

Before a widget is added, whether widget is attached or not is checked in `widget-open-handler.ts`. 
https://github.com/eclipse-theia/theia/blob/39887bf9d05c29fedc9466db76ca0a58706a1a2a/packages/core/src/browser/widget-open-handler.ts#L98-L100
However a widget is not necessarily attached immediately after the widget is added.
When we double click, click event is fired twice. The first event adds a widget and it can take some time to attach the widget for some reason (like network traffic?). In such a case, the second event can miss the widget exists and unnecessarily adds an additional widget.
This PR adds check for whether a widget exists or not, and prevents an additional widget from being added.

#### How to test


1.  Start browser example using Gitpod. (Maybe any other cloud IDE can be used.)
2. Open a file by double click.
3. Additional empty editor should not be opened.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
